modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/wal/crc/WalTestUtils.java

```diff
@@ -98,6 +103,78 @@ public static void corruptWalSegmentFile(
         }
     }
 
+    /**
+     * Put zero CRC in one of records for the specified compressed segment.
+     *
+     * @param desc WAL segment descriptor.
+     * @param pointer WAL pointer.
+     */
+    public static void corruptWalRecordInCompressedSegment(
+        FileDescriptor desc,
+        FileWALPointer pointer
+    ) throws IOException, IgniteCheckedException {
+        File tmp = Files.createTempDirectory("temp-dir").toFile();
+
+        U.unzip(desc.file(), tmp, new NullLogger());
+
+        File walFile = tmp.listFiles()[0];
+
+        String walFileName = desc.file().getName().replace(FilePageStoreManager.ZIP_SUFFIX, "");
+
+        // reanaming is needed because unzip removes leading zeros from archived wal segment file name,
+        // but strict name pattern of wal file is needed for WALIterator
+        walFile.renameTo(new File(tmp.getPath() + "/" + walFileName));
+
+        walFile = tmp.listFiles()[0];
+
+        final IgniteWalIteratorFactory factory = new IgniteWalIteratorFactory(new NullLogger());
+
+        IgniteWalIteratorFactory.IteratorParametersBuilder builder = new IgniteWalIteratorFactory.IteratorParametersBuilder()
+            .filesOrDirs(walFile);
+
+        try (WALIterator stIt = factory.iterator(builder)) {
+            while (stIt.hasNextX()) {
+                IgniteBiTuple<WALPointer, WALRecord> next = stIt.nextX();
+
+                final WALRecord record = next.get2();
+
+                if (pointer.equals(record.position())) {
+                    corruptWalRecord(new FileDescriptor(walFile), (FileWALPointer)next.get1());
+                    break;
+                }
+            }
+        }
+
+        byte[] fileBytes = U.zip(Files.readAllBytes(walFile.toPath()));
+
+        Files.write(desc.file().toPath(), fileBytes);
+
+        walFile.delete();
+
+        tmp.delete();
+    }
+
+    /**
+     * Corrupts zip file. According to section 4.3.7 of zip file specification, this method just sets 0
+     * to the first 30 bytes of the first local file header to achieve corruption.
+     *
+     * @param desc File descriptor.
+     * @see <a href="https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT">Zip file specification </a>
+     */
+    public static void corruptCompressedSegment(FileDescriptor desc) throws IOException {
```

Comment:
This method doesn't necessarily work with segments, it's applicable to any zip file.
Shouldn't we rename it?